app/web/features/search/FilterDialog.tsx

```diff
@@ -198,7 +198,7 @@ export default function FilterDialog({
                 id="last_active_filter"
                 className={classes.marginBottom}
                 value={lastActiveFilter}
-                onChange={(e) => setLastActiveFilter(e.target.value as number)}
+                onChange={(e) => setLastActiveFilter(parseInt(e.target.value as string))}
```

Comment:
Please update the type in `SearchPage` for this value rather than using `as string`. The reason it's giving you an error is that the `lastActiveFilter` and the `numberOfGuestsFilter` being passed to this component has a different type in `SearchBox`.

`as [TYPE]` should be used very rarely and usually signals something is defined wrong with the types.

Reply:
hmm, the thing is, I want the type to be a number, that's why I can use it in functions like `Number.isNan` or comparing if it's bigger than 0...
the `as string` I needed because (if I'm not wrong) because `e.target.value` is undefined, so I need to type it, and since it's an string I need to convert it to a number (if I'm following the conversation correctly)

Reply:
![image](https://github.com/user-attachments/assets/ff817587-f6f1-441a-869d-d8935e5576f4)

Reply:
basically TS complains because I can't convert an unknown typed variable to a number (even if it's not an unknown and its an actual string)

Reply:
Ahhh okay, I see what you mean.

Could you do it like this so it's more clear?

In reusable `Select` component add onChange prop with type like this:

```javascript
export default function Select({
    onChange
}: {
    onChange?: (event: React.ChangeEvent<{ value: unknown }>) => void;
}) =>

     <MuiSelect
        onChange={onChange}
      >
```

That makes it more clear it's supposed to be that way. Right now it's hidden in `otherProps`.

And would you mind doing the coercion in a `handleLastActiveFilterChange function like this?:

```javascript
const handleLastActiveChange = (
  event: React.ChangeEvent<{ value: unknown; name?: string }>
) => {
  const value = event.target.value as number // This way you're casting it to a new value rather than doing it to event.target.value directly
  setLastActiveFilter(value);
};
```

Reply:
ok, the thing is

I need to parse it to an number so I can compare it later, so `parseInt()` is something needed, if I do `as number` it's only for typescript right? but I need a number in JS the TS is only for the typing...

for using the parseInt it needs to receive an string, so if it's of type unknown complains, what I can do it's to modify the prop in the general select, to `event: React.ChangeEvent<{ value: string | unknown}>` I guess that would be fine? (feels to much chaning the prop just for this, but I guess it's the solution to avoid the `as string`)

Reply:
and this in the `handleLastActiveChange`, I guess with that it's fine? (since I need the `parseInt()`

Reply:
I think in this specific case it is a number, since the options you're giving the 1`Select` are numbers. The issue is the Material-UI `Select` type for the `onChange` says the value is unknown, and doesn't let us change it (I tried).

You can test this by logging `console.log('event.target.value type', typeof event.target.value)`  or `console.log('is event.target.value number?:, Number.isInteger(event.target.value)` and that should tell you the type.

Reply:
:D as I said, it's an string
![image](https://github.com/user-attachments/assets/1703b550-e0dd-4452-a3dc-798bbe98016c)

Reply:
I think I'm getting confused with the `hostingStatus` and `numGuests` 🙈.

Reply:
I think it's fine, I think all of this it's due an `as string` which I believe I need it, I suggest to merge this, and afterwards you can check if you find something improvable (but I want me who does it, if possible, since I want to improve by myself)

Reply:
also I feel we are blocking it bc of that, and this is a small PR

Reply:
@bakeiro happy to approve this one once it's passing the gitlab check!